sdks/python/apache_beam/io/filesystem.py

```diff
@@ -425,6 +433,22 @@ def _get_compression_type(path, compression_type):
                       'was %s' % type(compression_type))
     return compression_type
 
+  @classmethod
+  def get_all_subclasses(cls):
+    """Get all the subclasses of the FileSystem class
+    """
+    all_subclasses = []
+    for subclass in cls.__subclasses__():
+      all_subclasses.append(subclass)
+      all_subclasses.extend(subclass.get_all_subclasses())
```

Comment:
How performant is this (say, for looking up GCSFileSystem in current code) ? Should we cache results ?